File: easy/26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        write = 1
        read = 1
        while read < len(nums):
            if nums[write - 1] != nums[read]:
                nums[write] = nums[read]
                write += 1
            read += 1

        return write

# @lc code=end

File: easy/26.remove-duplicates-from-sorted-array.js
/*
 * @lc app=leetcode id=26 lang=javascript
 *
 * [26] Remove Duplicates from Sorted Array
 * [2pointers]
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  // ordered array and in-place modification => 2 pointers
  let write = 0;
  let read = 0;
  while(read < nums.length) {
    if (nums[write] === nums[read]) {
      read++;
    } else {
      nums[++write] = nums[read];
      read++;
    }
  }

  return write + 1;
};
// @lc code=end

